### PR TITLE
Add Android implementation

### LINKED ISSUE #146: Don't support android?
undefined is not an object

### PR DESCRIPTION
**Pull request by alpha0010**
Fixes #146

### CODE REVIEW

**Comment by alpha0010** [comment]
@iRoachie Any feedback on this?

**Comment by iRoachie** [comment]
@alpha0010 Thanks for the interest. Apologize, the notification got lost in my email. I'm not sure if we should add this or not. Right now the extent of the maintenance of this library was just bug fixes and making sure it was working with new react native versions. 

However, it wasn't really getting any active development as its only an implementation of UISearchBar which is pretty limited in itself. Also javascript only alternatives are up to scratch with the native ones in both design and performance. @umhan35 if you have any persons that you think can maintain this lib (active development) then you can let me know.

**Comment by alpha0010** [comment]
Our team has the strong goal to use the "real" widgets whenever possible.  JS widgets are great, especially for more complicated components.  However, there is always the chance an OS update will change the UX of a widget, making the JS versions of common base widgets loose some of the native feel.

I would guess others may have similar goals.

**Comment by iRoachie** [comment]
Hey @alpha0010 I tried running it with a new project and it failed for me. 

Steps I took to test:
- `react-native init MyNewProject`
- `cd MyNewProject`
- `npm install --save https://github.com/alpha0010/react-native-search-bar.git`
- `react-native link react-native-search-bar`
- `react-native run-android`

```
Task :app:transformClassesWithDexBuilderForDebug FAILED
```

Is there any additional steps needed to get this working?

**Comment by alpha0010** [comment]
Oops, forgot to mention.  You need to add

```gradle
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
```

to the `android` section in `app/build.gradle`.

**Comment by alpha0010** [comment]
@iRoachie Any progress on this?

**Comment by kvandake** [comment]
Any progress on this pull request?

**Comment by prathammehta** [comment]
Thanks for this repo! Would love to see this added.

**Comment by alpha0010** [comment]
@iRoachie Note that the above mentioned compile options are now default in the react native app.

**Comment by Noitidart** [comment]
I strongly support merging this please. Thank you @umhan35 for working on this! The native search bar is exactly what i needed, without the customization. The customization leads to it not looking native. The native search bars respect the native dark mode so that gives us all we need.

File: android/src/main/java/org/umhan35/RNSearchBarManager.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    }
+
+    @ReactProp(name = "text")
+    public void setText(SearchView searchView, String text) {
+        searchView.setQuery(text, false);
+    }
+
+    @ReactProp(name = "textColor", customType = "Color")
+    public void setTextColor(SearchView searchView, @Nullable Integer color) {
+        if (color != null) {
```
Comment (vonovak) [inline android/src/main/java/org/umhan35/RNSearchBarManager.java]: when someone specifies `textColor` prop and then resets it to `undefined`, then `color` here will be `null` and the JS prop change will be ignored

**Comment by Noitidart** [comment]
Thanks @vonovak - I'm really looking forward to this feature for my Android app! Currently I use this in ios, and then a cheesy js implemtation for android.

**Comment by iRoachie** [comment]
Is there anything needed to be added for react-native 0.60.0 autolinking support?

File: android/src/main/java/org/umhan35/RNSearchBarManager.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            default:
+                break;
+        }
+
+        searchView.setInputType(flags);
+    }
+
+    @ReactProp(name = "editable")
+    public void setEditable(SearchView searchView, boolean editable) {
+        if (!editable) {
```
[the remark this answers is not in the document]
Comment (alpha0010) [reply android/src/main/java/org/umhan35/RNSearchBarManager.java]: Whoops, over optimized.  Fixed now.

File: android/src/main/java/org/umhan35/RNSearchBarManager.java (hunk 1 above)
Comment (alpha0010) [reply android/src/main/java/org/umhan35/RNSearchBarManager.java]: Harder problem than it looks at first.  When undefined, the system default text color is used.  If we are explicitly setting it back to default, we must determine what said color was (awkward, but possible).  *However*, default system behavior changes the color of the text when it is disabled, but does *not* do so when we override via this function.

**Comment by alpha0010** [comment]
Autolinking works fine with this branch; no changes required.

File: android/src/main/java/org/umhan35/RNSearchBarManager.java (hunk 1 above)
Comment (alpha0010) [reply android/src/main/java/org/umhan35/RNSearchBarManager.java]: Figured it out. Fixed now.

**Comment by iRoachie** [comment]
Can you remove the index.js.flow?

**Comment by alpha0010** [comment]
Done.

**Comment by iRoachie** [comment]
Launched in [v3.5.0](https://github.com/umhan35/react-native-search-bar/releases/tag/v3.5.0). Thanks everyone for your hard work 💯

File: android/src/main/java/org/umhan35/RNSearchBarManager.java (hunk 1 above)
Comment (Noitidart) [reply android/src/main/java/org/umhan35/RNSearchBarManager.java]: Thanks @alpha0010 for your hard work and research here!!

**Comment by Noitidart** [comment]
@alpha0010 is there a way to make the "magnifying glass" turn into the native "arrow back" button? (pressing this will blur the field) (similar to the ios "Cancel")

And also is there a way to show the native "X" at the far right when there is text in there? (pressing this will clear the field) (similar to the ios "X")

**Pull request merged**